pkg/client/client_test.go

```diff
@@ -211,24 +269,47 @@ var _ = Describe("Client", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(cl).NotTo(BeNil())
 
+			By("initially creating a Deployment")
 			dep, err := clientset.AppsV1().Deployments(ns).Create(dep)
 			Expect(err).NotTo(HaveOccurred())
 
-			By("updating the object")
+			By("updating the Deployment")
 			dep.Annotations = map[string]string{"foo": "bar"}
 			err = cl.Update(context.TODO(), dep)
 			Expect(err).NotTo(HaveOccurred())
 
-			By("writing the result back to the go struct")
+			By("validating updated Deployment exists")
 			actual, err := clientset.AppsV1().Deployments(ns).Get(dep.Name, metav1.GetOptions{})
 			Expect(err).NotTo(HaveOccurred())
 			Expect(actual).NotTo(BeNil())
 
 			close(done)
 		})
 
-		It("should update an existing new object from an unstructured object", func() {
-			// TODO(seans3): implement these
+		It("should update an existing new object from an unstructured object", func(done Done) {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cl).NotTo(BeNil())
+
+			By("initially creating a Deployment")
+			dep, err := clientset.AppsV1().Deployments(ns).Create(dep)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating and encoding the Deployment as unstructured")
+			var u runtime.Unstructured = &unstructured.Unstructured{}
+			dep.Annotations = map[string]string{"foo": "bar"}
+			scheme.Convert(dep, u, nil)
+
+			By("updating the Deployment")
+			err = cl.Update(context.TODO(), u)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("fetching newly created unstructured Deployment")
+			actual, err := clientset.AppsV1().Deployments(ns).Get(dep.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
We aren't verifying the update actually happened.

Reply:
Doesn't this line do that:

> Expect(actual.Annotations["foo"]).To(Equal("bar"))

Reply:
I just pushed this commit to address Sunil's concern.